core/include/moveit/task_constructor/storage.h

```diff
@@ -325,15 +325,18 @@ class SolutionSequence : public SolutionBase
 public:
 	using container_type = std::vector<const SolutionBase*>;
 
-	explicit SolutionSequence() : SolutionBase() {}
-	SolutionSequence(container_type&& subsolutions, double cost = 0.0, Stage* creator = nullptr)
-	  : SolutionBase(creator, cost), subsolutions_(std::move(subsolutions)) {}
+	explicit SolutionSequence();
+
+	SolutionSequence(container_type&& subsolutions, double cost = 0.0, Stage* creator = nullptr);
 
 	void push_back(const SolutionBase& solution);
 
 	/// append all subsolutions to solution
 	void fillMessage(moveit_task_constructor_msgs::Solution& msg, Introspection* introspection) const override;
 
+	using CostAggregator = std::function<double(double, double)>;
```

Comment:
> this would also require to define a different neutral element (1 instead of 0)

I don't see why. Nothing in the current computation refers to a neutral element.

> Not sure we really need something else than addition

Neither am I, but I already implemented it for other reasons and I don't see the need to remove it.
It adds configurability for the user and it works well, e.g., with `std::max()`/`min()` (see the resp. test).

Reply:
> I don't see why. Nothing in the current computation refers to a neutral element.

Costs are initialized to zero here:
https://github.com/ros-planning/moveit_task_constructor/blob/675558219bf6f129433cda01a1334f8c10537ad4/core/src/storage.cpp#L234

However, multiplication will require a 1.0 as the neural element / initialization.

Reply:
That place is trivial to fix (unroll the first loop), but it turns out [there are others places where fixes would be more ugly](https://github.com/ros-planning/moveit_task_constructor/blob/master/core/include/moveit/task_constructor/container.h#L109).

I don't really like it, but I removed the aggregator support again. I do still think there would be merit in supporting, e.g. , `min`/`max` as operators. Multiplication is equivalent to addition up to a logarithm, so it does not change a lot in contrast.